Modules/_json.c

```diff
@@ -1720,11 +1763,8 @@ encoder_listencode_list(PyEncoderObject *s, _PyUnicodeWriter *writer,
     }
 
     if (s->indent != Py_None) {
-        Py_CLEAR(new_newline_indent);
-        Py_CLEAR(separator_indent);
-        if (_PyUnicodeWriter_WriteStr(writer, newline_indent) < 0) {
-            goto bail;
-        }
+        indent_level--;
+        do_dedent(s, writer, indent_level, indent_cache);
```

Comment:
The result value of `do_dedent` should be handled right?
```suggestion
        if (do_dedent(s, writer, indent_level, indent_cache) < 0) {
            goto bail;
        }
```

Reply:
Right.